infra/nightly-resources/index.js

```diff
@@ -1,46 +1,202 @@
+            commit: commit.slice(0, -1),
+            // type coerce a unix timestamp that is a string into an int with a `+`
+            date: +date,
+            // The file server/DOM parser constructs the absolute URL (of a relative URL) based on where we're requesting from,
+            // and since we're requesting from a sibling page, it mangles the link to look something like:
+            //   "/reports/eggcc/<current report>/<report requested>" 
+            // which is a nonsensical url. We can reconstruct the URL with the href attribute which we do here
+            url: `https://nightly.cs.washington.edu/reports/eggcc/${hrefText}`, 
+        }
+
+        comparisons.push(run);
+    }
+
+    comparisons.sort((l, r) => {
+        if (l.date < r.date) {return -1; }
+        if (l.date > r.date) {return 1; }
+        return 0;
+    });
+
+    return comparisons;
+}
+
+async function buildNightlyDropdown(element, previousRuns, initialIdx) {
+    const select = document.getElementById(element);
+    
+    const formatRun = (run) => `${run.branch} - ${run.commit}`
+    
+    previousRuns.forEach((nightly) => {
+        console.log(nightly);
+
+        const option = document.createElement("option");
+        option.innerText = formatRun(nightly);
+        select.appendChild(option);
+    });
+    
+    select.onchange = () => {
+        const compareTo = previousRuns[select.selectedIndex];
+        loadBenchmarks(compareTo);
+    }
+
+    select.selectedIndex = initialIdx;
+    select.value = formatRun(previousRuns[initialIdx]);
+    select.onchange();
+}
+
+// findBenchToCompare will find the last benchmark run on the main branch that is not itself
+function findBenchToCompareIdx(benchRuns) {
+    // Determine what benchmark run we are based on the browser's URL
+    // This is likely the best way to do this without embedding a bunch of data into our profile.js file
+    // or our profile.json file, which although tempting, is not backwards compatible
+    const path = window.location.pathname;
+    const parts = path.split("/");
+
+    // URLs should have trailing slashes leaving `parts` with a blank last element,
+    // so we should index into `parts` at its length-2
+    // Just in case the URL somehow doesn't have a trailing slash and `parts` doesn't
+    // have a blank last element, do a quick check and adjust the index accordingly
+    const idx = path[path.length - 1] === "/" ? parts.length - 2 : parts.length - 1;
+    
+    const [date, _, branch, commit] = parts[idx].split("%3A");
+    for (let i = benchRuns.length-1; i >= 0; i--) {
+        const run = benchRuns[i];
+        if (run.branch === "main") {
+            // If we are comparing a run on a main branch, to previous main branch we need to make sure
+            // it is not the same branch.
+            // I did mean `==` here, not `===`. `curComparison.date` is an int, and `date` is a string
```

Comment:
I think we should use `===` but cast it or convert it to an int explicitly

Reply:
Yeah, we could explicitly cast, but I'm not 100% sure I see the need here. Under the hood JS is doing the same thing, and I think that `==` is just as explicit as me writing something like:
```javascript
curComparision.date === +date // this one is a slight hack, but I like the look of it for non-negative numbers
```
or
```javascript
curComparision.date === parseInt(date, 10)
```

Reply:
Yeah I just hate `==` because idk wtf it does. whereas I can look at parseInt and understand.
But this is an optional fix